Content.Shared/Stunnable/SharedStunSystem.cs

```diff
@@ -248,13 +249,65 @@ public bool TrySlowdown(EntityUid uid, TimeSpan time, bool refresh,
             slowed.SprintSpeedModifier *= runSpeedMultiplier;
 
             _movementSpeedModifier.RefreshMovementSpeedModifiers(uid);
-
             return true;
         }
 
         return false;
     }
 
+    /// <summary>
+    /// Updates the movement speed modifiers of an entity by applying or removing the <see cref="SlowedDownComponent"/>.
+    /// If both walk and run modifiers are approximately 1 (i.e. normal speed) and <see cref="StaminaComponent.StaminaDamage"/> is 0,
+    /// or if the both modifiers are 0, the slowdown component is removed to restore normal movement.
+    /// Otherwise, the slowdown component is created or updated with the provided modifiers,
+    /// and the movement speed is refreshed accordingly.
+    /// </summary>
+    /// <param name="uid">Entity whose movement speed should be updated.</param>
+    /// <param name="walkSpeedModifier">New walk speed modifier. Default is 1f (normal speed).</param>
+    /// <param name="runSpeedModifier">New run (sprint) speed modifier. Default is 1f (normal speed).</param>
+    /// <param name="component">
+    /// Optional <see cref="StaminaComponent"/> of the entity.
+    /// </param>
+
+    public void UpdateStunModifiers(Entity<StaminaComponent?> uid,
```

Comment:
```suggestion
    public void UpdateStunModifiers(Entity<StaminaComponent?> ent,
```
it's not an uid, so please rename these to fit the naming scheme we use everywhere else
same for the other cases